src/pocketmine/entity/MobsControl.php

```diff
@@ -0,0 +1,368 @@
+		$this->healthMap["Default"] = 20;
+		$this->proximityMap["Default"] = 10;
+		$this->followMap["Default"] = MobsControl::FOLLOW_NORMAL;
+		$this->countMap["PlaceHolder"] = 0;
+		
+		
+// 		$this->speedMap["Zombie"] = .1;
+// 		$this->damageMap["Zombie"] = 1;
+// 		$this->healthMap["Zombie"] = 20;
+// 		$this->proximityMap["Zombie"] = 10;
+		
+// 		$this->speedMap["Creeper"] = .1;
+// 		$this->damageMap["Creeper"] = 1;
+// 		$this->healthMap["Creeper"] = 20;
+// 		$this->proximityMap["Creeper"] = 10;
+		
+// 		$this->speedMap["Skeleton"] = .1;
+// 		$this->damageMap["Skeleton"] = 1;
+// 		$this->healthMap["Skeleton"] = 20;
+// 		$this->proximityMap["Skeleton"] = 10;
+		
+// 		$this->speedMap["Spider"] = .1;
+// 		$this->damageMap["Spider"] = 1;
+// 		$this->healthMap["Spider"] = 20;
+// 		$this->proximityMap["Spider"] = 10;
+		
+// 		$this->speedMap["ZombiePigman"] = .1;
+// 		$this->damageMap["ZombiePigman"] = 1;
+// 		$this->healthMap["ZombiePigman"] = 20;
+// 		$this->proximityMap["ZombiePigman"] = 10;
+		
+// 		$this->speedMap["Slime"] = .1;
+// 		$this->damageMap["Slime"] = 1;
+// 		$this->healthMap["Slime"] = 20;
+// 		$this->proximityMap["Slime"] = 10;
+		
+// 		$this->speedMap["Enderman"] = .1;
+// 		$this->damageMap["Enderman"] = 1;
+// 		$this->healthMap["Enderman"] = 20;
+// 		$this->proximityMap["Enderman"] = 10;
+		
+// 		$this->speedMap["Silverfish"] = .1;
+// 		$this->damageMap["Silverfish"] = 1;
+// 		$this->healthMap["Silverfish"] = 20;
+// 		$this->proximityMap["Silverfish"] = 10;
+		
+// 		$this->speedMap["CaveSpider"] = .1;
+// 		$this->damageMap["CaveSpider"] = 1;
+// 		$this->healthMap["CaveSpider"] = 20;
+// 		$this->proximityMap["CaveSpider"] = 10;
+		
+// 		$this->speedMap["Ghast"] = .1;
+// 		$this->damageMap["Ghast"] = 1;
+// 		$this->healthMap["Ghast"] = 20;
+// 		$this->proximityMap["Ghast"] = 10;
+		
+// 		$this->speedMap["MagmaCube"] = .1;
+// 		$this->damageMap["MagmaCube"] = 1;
+// 		$this->healthMap["MagmaCube"] = 20;
+// 		$this->proximityMap["MagmaCube"] = 10;
+		
+// 		$this->speedMap["Blaze"] = .1;
+// 		$this->damageMap["Blaze"] = 1;
+// 		$this->healthMap["Blaze"] = 20;
+// 		$this->proximityMap["Blaze"] = 10;
+		
+// 		$this->speedMap["ZombieVillager"] = .1;
+// 		$this->damageMap["ZombieVillager"] = 1;
+// 		$this->healthMap["ZombieVillager"] = 20;
+// 		$this->proximityMap["ZombieVillager"] = 10;
+	}
+	
+	public function getstate(){
+		return $this->state;
+	}
+	
+	public function setstate($newstate) {
+		$this->state = $newstate;
+	}
+	
+	public function setSpeed($mobName, $speed) {
+		if ($mobName === "All") {
+			foreach($this->types as $val) {
+				$this->speedMap[$val] = $speed;
+			}
+			MainLogger::getLogger()->info("Set speed for all mobs to: $speed");
+		} else if (in_array($mobName, $this->types)) {
+			$this->speedMap[$mobName] = $speed;
+			MainLogger::getLogger()->info("Set $mobName speed to: $speed");
+		} else {
+			MainLogger::getLogger()->info("$mobName is not a supported mob");
```

Comment:
Not sure I follow.  Are you saying plugin wants to instantiate a mob and override the proximity, health, etc set by Mobs Control?  Is their an API now for plugins creating their own mob instance?  If so could it be updated to pass in criteria?